robottelo/ui/hosts.py

```diff
@@ -12,82 +12,98 @@ def _configure_hosts(self, arch=None, custom_ptable=None, cv=None,
+        strategy1, value1 = locators['host.select_puppetmodule']
+        strategy2, value2 = locators['host.select_puppetclass']
         # Host tab
-        if lifecycle_env is not None:
+        if lifecycle_env:
             Select(
                 self.wait_until_element(locators['host.lifecycle_env'])
             ).select_by_visible_text(lifecycle_env)
-        if cv is not None:
+        if cv:
             Select(
                 self.wait_until_element(locators['host.cv'])
             ).select_by_visible_text(cv)
-        if host_group is not None:
+        if reset_puppetenv:
+            self.click(locators['host.reset_puppetenv'])
+        if host_group:
             Select(
                 self.wait_until_element(locators['host.group'])
             ).select_by_visible_text(host_group)
-        elif env is not None and host_group is None:
+        elif env and host_group is None:
             # As selecting hostgroup changes env
             Select(
                 self.wait_until_element(locators['host.environment'])
             ).select_by_visible_text(env)
-        if puppet_ca is not None:
+        if puppet_ca:
             Select(
                 self.wait_until_element(locators['host.puppet_ca'])
             ).select_by_visible_text(puppet_ca)
-        if puppet_master is not None:
+        if puppet_master:
             Select(
                 self.wait_until_element(locators['host.puppet_master'])
             ).select_by_visible_text(puppet_master)
         # Network tab
-        self.wait_until_element(tab_locators['host.tab_network']).click()
-        if domain is not None:
+        if domain or mac or subnet or ip_addr:
+            self.click(tab_locators['host.tab_network'])
+        if domain:
             Select(
                 self.wait_until_element(locators['host.domain'])
             ).select_by_visible_text(domain)
-        if mac is not None:
+        if mac:
             self.wait_until_element(locators['host.mac']).send_keys(mac)
-        if subnet is not None:
+        if subnet:
             Select(
                 self.wait_until_element(locators['host.subnet'])
             ).select_by_visible_text(subnet)
-        if ip_addr is not None:
+        if ip_addr:
             self.wait_until_element(locators['host.ip']).send_keys(ip_addr)
         # Operating system tab
-        self.wait_until_element(tab_locators['host.tab_os']).click()
-        if arch is not None:
+        if arch or os or media or ptable or custom_ptable or root_pwd:
+            self.click(tab_locators['host.tab_os'])
+        if arch:
             Select(
                 self.wait_until_element(locators['host.arch'])
             ).select_by_visible_text(arch)
             self.wait_for_ajax()
-        if os is not None:
+        if os:
             Select(
                 self.wait_until_element(locators['host.os'])
             ).select_by_visible_text(os)
             self.wait_for_ajax()
```

Comment:
Can `click` be used here? I've seen it replace `wait_for_ajax` quite a bit lately.